hawk/src/main/java/com/orhanobut/hawk/AesEncryption.java

```diff
@@ -71,28 +70,18 @@ public void reset() {
      * Gets the secret key by using salt and password. Salt is stored in the storage
      * If the salt is not stored, that means it is first time and it creates the salt and
      * save it in the storage
-     *
-     * @return the secret key
      */
-    private AesCbcWithIntegrity.SecretKeys getSecretKey() {
-        AesCbcWithIntegrity.SecretKeys key = null;
+    private void generateSecretKey(String password) {
         try {
-            if (!TextUtils.isEmpty(saltKey)) {
-                key = AesCbcWithIntegrity.generateKeyFromPassword(PASSWORD, saltKey);
+            if (TextUtils.isEmpty(saltKey)) {
+                saltKey = AesCbcWithIntegrity.saltString(AesCbcWithIntegrity.generateSalt());
+                storage.put(KEY_SALT, saltKey);
             }
 
-            // already generated
-            if (key != null) {
-                return key;
-            }
-            saltKey = AesCbcWithIntegrity.saltString(AesCbcWithIntegrity.generateSalt());
-            key = AesCbcWithIntegrity.generateKeyFromPassword(PASSWORD, saltKey);
-            storage.put(KEY_SALT, saltKey);
+            key = AesCbcWithIntegrity.generateKeyFromPassword(password, saltKey);
         } catch (GeneralSecurityException e) {
-            Logger.d(e.getMessage());
+            throw new RuntimeException(e);
```

Comment:
This can crash the apps which uses the library, I do not know what should be the correct behavior here